CMakeLists.txt

```diff
@@ -182,6 +183,23 @@ if(SPHINX_FOUND)
   )
 endif()
 
+# =============================================================================
+# Build full docs
+# =============================================================================
+if(DOXYGEN_FOUND AND SPHINX_FOUND)
+  add_custom_target(
+          docs
+          COMMENT "Generating full documentation"
+  )
+  add_dependencies(docs doxygen prepare_docs sphinx)
```

Comment:
the dependencies are built out of order, so for example `prepare_docs` may be done after sphinx leading to missing notebooks.